### ISSUE TITLE
.Take(0) after .Skip causes an SqlException on SQL Server

### ISSUE THREAD

**Issue by MarkFl12** [dotnet/efcore#22525]
If you do the following: 

```
db.Articles
	.OrderBy(a => a.Published)
	.Skip(0)
	.Take(0)
	.ToList();
```

then it generates a query using ```OFFSET @__p_0 ROWS FETCH NEXT @__p_0 ROWS ONLY``` which with a zero in Take gives the error ```The number of rows provided for a FETCH clause must be greater then zero.```

If you were to perform this same query in old non-Core Entity Framework you get the expected zero rows returned to you.

### Further technical details

EF Core version: 3.1.8
Database provider: Microsoft.EntityFrameworkCore.SqlServer
Target framework: .NET Core 3.1
Operating system: Windows 10
IDE: Visual Studio 2019 16.7.3

**Comment by roji** [comment]
Sqlite and PostgreSQL are fine with OFFSET 0/LIMIT 0, this seems to be an SQL Server-specific limitation.

**Comment by roji** [comment]
> If you were to perform this same query in old non-Core Entity Framework you get the expected zero rows returned to you.

Note that SQL Server errors on both constant and parameter 0, so EF6 must have sniffed and eliminated the clauses (I've not confirmed this).

```sql
-- Constant
SELECT 1 AS x ORDER BY x OFFSET 0 ROWS FETCH NEXT 0 ROWS ONLY;

-- Parameterized
DECLARE @p INT = 0;
SELECT 1 AS x ORDER BY x OFFSET @p ROWS FETCH NEXT @p ROWS ONLY;
```

**Comment by MarkFl12** [comment]
Just tried it again on EF6 and it seems to shortcut out in the case of ```.Take(0)``` and it generates the following SQL:
```
SELECT 
    CAST(NULL AS int) AS [C1], 
    CAST(NULL AS datetime2) AS [C2], 
    CAST(NULL AS varchar(1)) AS [C5], 
    <removed some>
    CAST(NULL AS bit) AS [C19], 
    CAST(NULL AS int) AS [C21]
    FROM  ( SELECT 1 AS X ) AS [SingleRowTable1]
    WHERE 1 = 0
```

Seems to be some kind of detection for this situation in EF6 that hasn't been translated over to EF Core?

**Comment by GioviQ** [comment]
I have the same issue with EF 5.0.7.

**Comment by roji** [comment]
@GioviQ this was fixed for 6.0.0-preview1 (see the milestone above). Not sure this is a good candidate for servicing /cc @smitpatel.
